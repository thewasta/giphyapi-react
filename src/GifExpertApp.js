import React, {useState} from "react";
import {AddCategory} from "./components/AddCategory";
import GifGrid from "./components/GifGrid";

const GifExpertApp = () => {
    const [categories, setCategories] = useState(["Netflix"]);
    return (
        <>
            <div className="animate__animated animate__fadeIn">Expert App</div>
            <AddCategory setCategories={setCategories}/>
            <hr/>
            <ol>
                {
                    categories.map(category => (
                        <GifGrid
                            key={category}
                            category={category}/>
                    ))
                }
            </ol>
        </>
    );
};

export default GifExpertApp;