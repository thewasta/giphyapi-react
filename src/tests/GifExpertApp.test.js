import React from "react";
import {shallow} from "enzyme";
import GifExpertApp from "../GifExpertApp";

describe("Pruebas de Index", () => {
    test("Comprobar el snapshot", () => {
        const wrapper = shallow(<GifExpertApp/>);
        expect(wrapper).toMatchSnapshot();
    });
    test("debe mostrar lista de categor√≠as", function (assert) {
        const categories = ["One Punch", "Doraemon"];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find("GifGrid").length).toBe(categories.length);
        assert();
    });
});