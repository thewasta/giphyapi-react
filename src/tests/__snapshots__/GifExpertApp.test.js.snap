// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de Index Comprobar el snapshot 1`] = `
<Fragment>
  <div
    className="animate__animated animate__fadeIn"
  >
    Expert App
  </div>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas de Index debe mostrar lista de categor√≠as 1`] = `
<Fragment>
  <div
    className="animate__animated animate__fadeIn"
  >
    Expert App
  </div>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Doraemon"
      key="Doraemon"
    />
  </ol>
</Fragment>
`;
